### PR TITLE
Extend find all tasks filters

### PR DESCRIPTION
**Pull request by AdrianSad**
Goal of this MR is allowing the following filters:

- Parameter Id - text input
- 
- Failure date-time from / to
- 
- Created date-time from / to
- 
- Task Name - text input
- 
- Error Message contains text case insensitive (text input)

### CODE REVIEW

**Review by oliverselinger — CHANGES_REQUESTED** [review]

File: src/main/java/os/failsafe/executor/TaskRepository.java [hunk 1]
```diff
@@ -222,6 +223,39 @@ List<Task> findAll(String taskName, String parameter, Boolean failed, int offset
         return database.selectAll(sql, this::mapToPersistentTask, whereClause.params);
     }
 
+
+    public List<Task> findAll(String taskName, String parameter, Boolean failed, String errorMessage, LocalDateTime createdDateFrom, LocalDateTime createdDateTo,
```
[the remark this answers is not in the document]
Comment (AdrianSad) [reply src/main/java/os/failsafe/executor/TaskRepository.java]: changed

File: src/main/java/os/failsafe/executor/TaskRepository.java [hunk 2]
```diff
@@ -222,6 +223,39 @@ List<Task> findAll(String taskName, String parameter, Boolean failed, int offset
...
+
+        if (sorts.length == 0) {
+            sorts = new Sort[]{new Sort(Sort.Field.CREATED_DATE, Sort.Direction.DESC), new Sort(Sort.Field.ID, Sort.Direction.DESC)};
+        }
+
+        WhereBuilder.WhereClause whereClause = new WhereBuilder(database)
+                .where(taskName, "NAME")
+                .where(parameter, "PARAMETER")
+                .isNullOrNotNull(failed, "FAIL_TIME")
+                .gt(getTimeString(createdDateFrom), "CREATED_DATE")
```
[the remark this answers is not in the document]
Comment (AdrianSad) [reply src/main/java/os/failsafe/executor/TaskRepository.java]: changed

File: src/main/java/os/failsafe/executor/TaskRepository.java [hunk 3]
```diff
@@ -222,6 +223,39 @@ List<Task> findAll(String taskName, String parameter, Boolean failed, int offset
...
+                .lt(getTimeString(createdDateTo), "CREATED_DATE")
+                .gt(getTimeString(failureDateFrom), "FAIL_TIME")
```
[the remark this answers is not in the document]
Comment (AdrianSad) [reply src/main/java/os/failsafe/executor/TaskRepository.java]: changed

File: src/main/java/os/failsafe/executor/TaskRepository.java [hunk 4]
```diff
@@ -382,4 +416,10 @@ ExecutionFailure mapToExecutionFailure(ResultSet rs) throws SQLException, IOExce
 
         return new ExecutionFailure(failTime.toLocalDateTime(), exceptionMessage, stackTrace);
     }
+
+    private static String getTimeString(LocalDateTime time) {
+        return Optional.ofNullable(time)
+                .map(LocalDateTime::toString)
```
[the remark this answers is not in the document]
Comment (AdrianSad) [reply src/main/java/os/failsafe/executor/TaskRepository.java]: changed to `.map(Timestamp::valueOf)`

**Review by oliverselinger — APPROVED** [review]

**Pull request merged**